src/client/Client.cc

```diff
@@ -14970,7 +14970,7 @@ void Client::ms_handle_remote_reset(Connection *con)
       mds_rank_t mds = MDS_RANK_NONE;
       MetaSession *s = NULL;
       for (auto &p : mds_sessions) {
-	if (mdsmap->get_addrs(p.first) == con->get_peer_addrs()) {
+	if (mdsmap->have_inst(p.first) && mdsmap->get_addrs(p.first) == con->get_peer_addrs()) {
 	  mds = p.first;
 	  s = &p.second;
 	}
```

Comment:
I know this is old code and you are just adding a check, but it'd be good if the `first` and `second` references were converted to structured binding.

Reply:
you probably just need to move `mds` and `s` to the `for` loop initializer so that the `have_inst()` and `get_addrs()` calls look more sensible.

Reply:
Agree. But the whole client is littered with such .firs/.second semantics. I don't mind converting this to use structured binding, but we will still have the problem at large.

Reply:
yeah, I know.
I was just trying to nudge you to do that so that we could get the changes done bit by bit in relevant PRs rather than one huge PR to move the Client to structured bindings   :)

Reply:
I think it's safe to leave this as-is for now. But Milind is right it'd be nice to do that but it should go in a cleanup PR.